src/main/java/com/sst/utopia/booking/model/Ticket.java

```diff
@@ -0,0 +1,152 @@
+	 */
+	@EmbeddedId
+	private SeatLocation id;
+	/**
+	 * What class of seat this is. For now first class is "1", business class is
+	 * "2", and economy is "3". TODO: make an enum?
+	 */
+	@Column(name = "class")
+	private int seatClass;
+	/**
+	 * The holder of this ticket, if any.
+	 */
+	@Column(nullable = true)
+	private User reserver;
+	/**
+	 * The price that the ticket-holder paid to book this seat; must be null if
+	 * reserver is null or if both reserver and reservationTimeout are not null, but
+	 * must not be null if reserver is not null but reservationTimeout is.
+	 */
+	@Column(nullable = true)
+	private Integer price;
+	/**
+	 * If not null, how long the user who reserved this seat has to confirm that
+	 * reservation by paying for the ticket. Must be null if reserver is null or if
+	 * both reserver and price are not null, but must not be null if reserver is not
+	 * null but price is.
+	 */
+	@Column(nullable = true)
+	private LocalDateTime reservationTimeout;
+
+	/**
+	 * @return the flight, row, and seat that together uniquely identify this
+	 *         ticket.
+	 */
+	public SeatLocation getId() {
+		return id;
+	}
+
+	/**
+	 * @return the class of the seat
+	 */
+	public int getSeatClass() {
+		return seatClass;
+	}
+
+	/**
+	 * @return the ticket-holder, if any
+	 */
+	public User getReserver() {
+		return reserver;
+	}
+
+	/**
+	 * @return the price the ticket-holder paid, if the ticket is reserved and he or
+	 *         she has paid
+	 */
+	public Integer getPrice() {
+		return price;
+	}
+
+	/**
+	 * @return when the reservation will time out if the user who reserved the
+	 *         ticket has not paid by then
+	 */
+	public LocalDateTime getReservationTimeout() {
+		return reservationTimeout;
+	}
+
+	/**
+	 * If the ticket-holder is set to null, price and reservationTimeout are also
+	 * set to null.
+	 *
+	 * @param reserver the new ticket-holder
+	 */
+	public void setReserver(final User reserver) {
+		if (reserver == null) {
+			price = null;
+			reservationTimeout = null;
+		}
+		this.reserver = reserver;
+	}
+
+	/**
+	 * This method throws if the new price is not null but reserver is; if neither
+	 * reserver nor the new price are null, reservationTimeout is set to null.
+	 *
+	 * @param price the price the ticket-holder paid for the seat
+	 * @throws IllegalStateException if price is not null but reserver is
+	 */
+	public void setPrice(final Integer price) {
+		if (reserver == null && price != null) {
+			throw new IllegalStateException("Ticket can only be paid for if someone reserved it");
+		} else if (reserver != null && price != null) {
+			reservationTimeout = null;
+		}
+		this.price = price;
+	}
+
+	/**
+	 * This method throws if the new timeout is not null but reserver is.
+	 * @param reservationTimeout when the reservation will expire if not paid for.
+	 */
+	public void setReservationTimeout(final LocalDateTime reservationTimeout) {
+		if (reserver == null && reservationTimeout != null) {
```

Comment:
I am still confused as to how this works if the timeout has not been set yet.

Reply:
As with `setPrice`, `reservationTimeout` here is the parameter to the method, not the field in the object. But on reflection this should probably also clear `price`, or object if `price` has already been set.

Reply:
Yes, you are right.